packages/datadog-instrumentations/src/mongodb-core.js

```diff
@@ -0,0 +1,188 @@
+  Server.Server.prototype.command = wrapCommand(Server.Server.prototype.command, 'command')
+  Server.Server.prototype.insert = wrapCommand(Server.Server.prototype.insert, 'insert', 'insert')
+  Server.Server.prototype.update = wrapCommand(Server.Server.prototype.update, 'update', 'update')
+  Server.Server.prototype.remove = wrapCommand(Server.Server.prototype.remove, 'remove', 'remove')
+
+  Server.Cursor.prototype._getmore = wrapCursor(Server.Cursor.prototype._getmore, 'getMore', 'getMore')
+  Server.Cursor.prototype._find = wrapQuery(Server.Cursor.prototype._find, '_find')
+  Server.Cursor.prototype.kill = wrapCursor(Server.Cursor.prototype.kill, 'killCursors', 'killCursors')
+  return Server
+})
+
+addHook({ name: 'mongodb', versions: ['>=4'], file: 'lib/cmap/connection.js' }, Connection => {
+  const proto = Connection.Connection.prototype
+  proto.command = wrapConnectionCommand(proto.command, 'command')
+  proto.query = wrapConnectionCommand(proto.query, 'query')
+  proto.getMore = wrapConnectionCommand(proto.getMore, 'getMore', 'getMore')
+  proto.killCursors = wrapConnectionCommand(proto.killCursors, 'getMore', 'getMore')
+  return Connection
+})
+
+addHook({ name: 'mongodb', versions: ['>=3.3 <4'], file: 'lib/core/wireprotocol/index.js' }, wp => {
+  patch(wp)
+  return wp
+})
+
+addHook({ name: 'mongodb-core', versions: ['>=3.2'], file: 'lib/wireprotocol/index.js' }, wp => {
+  patch(wp)
+  return wp
+})
+
+addHook({ name: 'mongodb-core', versions: ['~3.1.10'], file: 'lib/wireprotocol/3_2_support.js' }, WireProtocol => {
+  WireProtocol.prototype.command = wrapUnifiedCommand(WireProtocol.prototype.command, 'command')
+  return WireProtocol
+})
+
+addHook({ name: 'mongodb-core', versions: ['~3.1.10'], file: 'lib/wireprotocol/2_6_support.js' }, WireProtocol => {
+  WireProtocol.prototype.command = wrapUnifiedCommand(WireProtocol.prototype.command, 'command')
+  return WireProtocol
+})
+
+addHook({ name: 'mongodb', versions: ['>=3.5.4'], file: 'lib/utils.js' }, util => {
+  shimmer.wrap(util, 'maybePromise', maybePromise => function (parent, callback, fn) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+    const callbackIndex = arguments.length - 2
+
+    callback = arguments[callbackIndex]
+
+    if (typeof callback === 'function') {
+      arguments[callbackIndex] = asyncResource.bind(callback)
+    }
+
+    return maybePromise.apply(this, arguments)
+  })
+  return util
+})
+
+function patch (wp) {
+  wp.command = wrapUnifiedCommand(wp.command, 'command')
+  wp.insert = wrapUnifiedCommand(wp.insert, 'insert', 'insert')
+  wp.update = wrapUnifiedCommand(wp.update, 'update', 'update')
+  wp.remove = wrapUnifiedCommand(wp.remove, 'remove', 'remove')
+  wp.query = wrapUnifiedCommand(wp.query, 'query')
+  wp.getMore = wrapUnifiedCommand(wp.getMore, 'getMore', 'getMore')
+  wp.killCursors = wrapUnifiedCommand(wp.killCursors, 'killCursors', 'killCursors')
```

Comment:
These should app be doing `shimmer.wrap()` like `shimmer.wrap(wp, 'killCursors', .......)`.